src/main/kotlin/net/atos/zac/signalering/SignaleringenJob.kt

```diff
@@ -0,0 +1,377 @@
+    private val zoekenService: ZoekenService,
+    private val takenService: TakenService
+) {
+
+    companion object {
+        private val LOG: Logger = Logger.getLogger(SignaleringenJob::class.java.name)
+        const val ZAAK_AFGEHANDELD_QUERY: String = "zaak_afgehandeld"
+    }
+
+    fun signaleringenVerzenden() {
+        zaakSignaleringenVerzenden()
+        taakSignaleringenVerzenden()
+    }
+
+    /**
+     * This is the batchjob to send E-Mail warnings about cases that are approaching their einddatum gepland or their
+     * uiterlijke einddatum afdoening.
+     */
+    fun zaakSignaleringenVerzenden() {
+        val info = SignaleringVerzendInfo()
+        LOG.info("Zaak signaleringen verzenden: gestart...")
+        ztcClientService.listZaaktypen(configuratieService.readDefaultCatalogusURI())
+            .forEach(
+                Consumer<ZaakType> { zaaktype: ZaakType ->
+                    val parameters = zaakafhandelParameterService.readZaakafhandelParameters(
+                        URIUtil.parseUUIDFromResourceURI(zaaktype.url)
+                    )
+                    if (parameters.einddatumGeplandWaarschuwing != null) {
+                        info.streefdatumVerzonden += zaakEinddatumGeplandVerzenden(
+                            zaaktype,
+                            parameters.einddatumGeplandWaarschuwing
+                        )
+                        zaakEinddatumGeplandOnterechtVerzondenVerwijderen(
+                            zaaktype,
+                            parameters.einddatumGeplandWaarschuwing
+                        )
+                    }
+                    if (parameters.uiterlijkeEinddatumAfdoeningWaarschuwing != null) {
+                        info.fataledatumVerzonden += zaakUiterlijkeEinddatumAfdoeningVerzenden(
+                            zaaktype,
+                            parameters.uiterlijkeEinddatumAfdoeningWaarschuwing
+                        )
+                        zaakUiterlijkeEinddatumAfdoeningOnterechtVerzondenVerwijderen(
+                            zaaktype,
+                            parameters.uiterlijkeEinddatumAfdoeningWaarschuwing
+                        )
+                    }
+                }
+            )
+        LOG.info(
+            """Zaak signaleringen verzenden: gestopt (${info.streefdatumVerzonden} streefdatum waarschuwingen,
+            | ${info.fataledatumVerzonden} fatale datum waarschuwingen)
+            """.trimMargin()
+        )
+    }
+
+    /**
+     * Send the E-Mail warnings about the einddatum gepland
+     *
+     * @return the number of E-Mails sent
+     */
+    fun zaakEinddatumGeplandVerzenden(zaaktype: ZaakType, venster: Int): Int {
+        val verzonden = IntArray(1)
+        zoekenService.zoek(getZaakSignaleringTeVerzendenZoekParameters(DatumVeld.ZAAK_STREEFDATUM, zaaktype, venster))
+            .items.stream()
+            .map { zaakZoekObject: ZoekObject? -> zaakZoekObject as ZaakZoekObject? }
```

Comment:
should this really be a nullable type? should you not change the `buildZaakSignalering` function maybe?